packages/runtime/container-runtime/src/dataStores.ts

```diff
@@ -430,31 +430,39 @@ export class DataStores implements IDisposable {
         );
     }
 
-    public async getDataStore(id: string, wait: boolean, viaHandle: boolean): Promise<FluidDataStoreContext> {
-        const context = await this.contexts.getBoundOrRemoted(id, wait);
+    public async getDataStore(id: string, requestHeaderData: RuntimeHeaderData): Promise<FluidDataStoreContext> {
+        const headerData = { ...defaultRuntimeHeaderData, ...requestHeaderData };
+
+        const context = await this.contexts.getBoundOrRemoted(id, headerData.wait);
         const request = { url: id };
         if (context === undefined) {
             // The requested data store does not exits. Throw a 404 response exception.
             throw responseToException(create404Response(request), request);
         }
 
         if (context.tombstoned) {
+            const shouldFail = this.throwOnTombstoneLoad && !headerData.allowTombstone;
+
             // The requested data store is removed by gc. Create a 404 gc response exception.
-            const error = responseToException(createResponseError(404, "Datastore removed by gc", request), request);
+            const error = responseToException(createResponseError(
+                404,
+                "Datastore removed by gc",
```

Comment:
We should rename this to "Datastore was deleted". No need to mention "gc" here as it sounds like gc did something wrong :-).

Reply:
(Pending)